src/configureWorkspace/scaffolding.ts

```diff
@@ -48,6 +49,7 @@ export type ScaffoldFile = {
     contents: string;
     fileName: string;
     open?: boolean;
+    createNewNameIfExist?: boolean;
```

Comment:
It seems conceivable that we may end up adding more configurable behavior on name conflicts (e.g. `promptForOverwrite`) which would end up complicating this type (e.g. which property would have precedence?) . It might be better to consolidate that behavior into a single "conflict resolution method" handler.

One option would be a simple enumeration:

```typescript
type ConflictResolutionMethod = 'prompt' | 'overwrite' | 'unique-name'
type ScaffoldFile = {
   ...
   onConflict?: ConflictResolutionMethod; // If omitted, defaults to 'prompt'
};
```

Another option would be to allow the individual scaffolders the flexibility to do whatever is needed (as it may be difficult to use a single unique naming scheme across all types, or the scaffolder type may want to use a more specific prompt to overwrite).  Something like:

```typescript
type ScaffoldFile = {
   ...
   onConflict?: (path: string, defaultHandler: (path: string): => Promise<string>) => Promise<string>;
};
```

In this case, the scaffolding type can attach a callback which the scaffolder will invoke, which will return the (possibly new) filename to use.  Also passed could be the "default" handler (that would normally prompt for overwrite) that could be called if the specific handler wanted a fallback in certain cases.

In either case, we might also consider how a scaffolder type could indicate that a file be "skipped" altogether when there's a conflict.

Reply:
I think the second suggestion is more open. But Why not use
`  onConflict?: (path: string) => Promise<string|undefined>;`

Since OnConflict is optional, then the default behavior of overwrite can be used if onConflict is undefined. No need to pass defaultHandler to callback.

Reply:
My original thinking was that returning `undefined` didn't make clear (or allow for) whether the default behavior should be used (e.g. prompt for overwrite) or whether the file should just be skipped (as you suggest).  In the latter case, the platform-specific scaffolder is then responsible for prompting for override, if that's ultimately the behavior they want for that file in that case.  (Passing the default allows the scaffolder the option to perform completely custom logic, or some combination of custom and default behavior.)

It only really becomes an issue if/when that default behavior gets more sophisticated.  Since, today, it only prompts for overwriting the file, it may be overkill to add that extra flexibility.

Reply:
I think this should work for most cases and can be improved when needed.